src/store/transactions/transaction.hooks.ts

```diff
@@ -260,3 +270,34 @@ export const useLocalTransactionInputsState = () => useAtom(localStacksTransacti
 export const useLocalTransactionState = () => useAtom(localTransactionState);
 export const useTxForSettingsState = () => useAtom(txForSettingsState);
 export const useTxByteSizeState = () => useAtom(txByteSize);
+
+export const useTokenRealAmount = (tx: ContractCallTransaction) => {
+  const asset = useAssetByContractId(tx?.contract_call?.contract_id);
+  if (!isContractCallWithTokenTransfers(tx) || !asset) return;
+  const amount = calculateTokenTransferAmount(asset, tx);
+  if (typeof amount === 'undefined') return;
+  return amount.toNumber().toLocaleString('en-US', { maximumFractionDigits: 2 });
```

Comment:
Thomas' point on duplicate was valid, but this leaves us no way to consume the actual value, so I'd recommended breaking apart the calculation & formatting. This is important to avoid code like we had before (`removeCommas` to try and get a number type out of a formatted value)

Reply:
calculateTokenTransferAmount only does the calculation (it returns a BigNumber)
Then we format at line 279
I will then add another utility method that only does the formatting (basically on toLocaleString('en-US', { maximumFractionDigits: 2 }))

Reply:
Yep makes sense. I guess my point is also that formatting of a number is a presentational concern, and not something the store needs to have knowledge of, so you could use this utility in the component itself.

Reply:
Got it , I will add it there !

Reply:
I added it to useTxValue which does all the formatting and is used in the component.
https://github.com/blockstack/stacks-wallet-web/blob/feat/show-contract-activity/I1460/src/store/transactions/transaction.hooks.ts#L285